experimental/calendar/src/duration.rs

```diff
@@ -0,0 +1,60 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::Calendar;
+use std::fmt;
+use std::marker::PhantomData;
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+/// A duration between two dates
+pub struct DateDuration<C: Calendar + ?Sized> {
+    pub years: i32,
+    pub months: i32,
+    pub weeks: i32,
+    pub days: i32,
+    pub marker: PhantomData<C>,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum DurationUnit {
+    Years,
+    Months,
+    Weeks,
+    Days,
+}
+
+impl<C: Calendar + ?Sized> Default for DateDuration<C> {
+    fn default() -> Self {
+        Self {
+            years: 0,
+            months: 0,
+            weeks: 0,
+            days: 0,
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<C: Calendar + ?Sized> DateDuration<C> {
+    pub fn new(years: i32, months: i32, weeks: i32, days: i32) -> Self {
+        DateDuration {
+            years,
+            months,
+            weeks,
+            days,
+            marker: PhantomData,
+        }
+    }
+}
```

Comment:
Suggestion: maybe we don't need `new` since we can construct the struct directly in Rust

Reply:
The marker makes it annoying and I'd prefer if users didn't have to care about the marker